### DISCUSSION TITLE
TestFlight Version 1.4.4 (AppStore Version)

### DISCUSSION THREAD

**Discussion by hape42** [hape42/dailygammon#85] [General]
When you describe problems, the device and the IOS version should always be mentioned.

minimum feedback for all should be something like: "tested on iPhone XR IOS 16.1 no problems found so far".

thank you and happy testing

**Comment by hape42** [comment]
I am testing with an i**Phone XR** IOS **16.1.2** and **iPad 7** IOS **15.6.1**

No problems so far.

**Reply by hape42** [reply]
The TopPage button does not work anymore. 

I have already found the error and fixed it.

**Comment by Smiles2go** [comment]
Testing new build 336, iPhone 14pro iOS 16.2. 

Issue: Settings screen layout gets stretched such that some settings aren't visible. Screenshot:
![image](https://user-images.githubusercontent.com/121371957/212457359-8ae95543-07b9-4706-a7ce-2d1b347f3928.png)

I suspected IOS settings/display&brightness/Display Zoom, which I like on. So I turned off Zoom and now can see all the settings:

![image](https://user-images.githubusercontent.com/121371957/212459261-f6fa54aa-e46a-4bbc-8bf0-5b97e1f0c564.png)

**Reply by hape42** [reply]
Thanks for the tip, I'll take care of it right away.

**Reply by Smiles2go** [reply]
Thanks. More inconvenience than really bug, but Top Page only displays 3 matches in Zoom mode but 5 in default:
![image](https://user-images.githubusercontent.com/121371957/212476879-003c87b1-be03-4c69-9648-0c0da8cf66dd.png)

**Reply by Smiles2go** [reply]
Some button text contrast has suffered. More contrast would be appreciated in general. Here is an example that used to be easier to read (Next and To Top buttons):

![image](https://user-images.githubusercontent.com/121371957/212476988-4b4ae172-59fa-4045-9ba6-4378d1ec9fc3.png)

**Reply by hape42** [reply]
Hmmm

If you want to make everything bigger on the same available space (zoom mode) this contradicts with "show as much information as possible without scrolling".

If I would make the lines closer together, they would be difficult to hit with the finger in normal mode.

The zoom mode would make the font so large that only 3 - 4 lines would be displayed.

It is difficult for the developers to satisfy everyone at the same time. Here you will have to live with a few compromises to handle different zoom sizes at the same time.

Translated with www.DeepL.com/Translator (free version)

**Reply by hape42** [reply]
Some boards do not harmonize optimally with every button.

Everyone can now make the optimal setting for themselves.

That all boards with all buttons harmonize optimally is certainly possible. But we would need more designers and programmers and time.

**Reply by b-dubbs** [reply]
Love the app and your improvements.  I was just about to post a very similar comment.  The contrast in the buttons for this board (my favorite board) on the 'next game/to top' dialog after a move DID suffer.  

IF POSSIBLE - a relatively simple fix for those buttons would be to use the same background for those buttons as is used for other buttons in the same board configuration.  All of the other buttons have the grey background and have enough contrast for a visually impaired person to use.  However, this configuration -- with the light/white background doesn't work well.

Thanks again for hearing us and for the app -- WONDERFUL work :)

**Reply by Smiles2go** [reply]
> Hmmm
> 
> 
> 
> If you want to make everything bigger on the same available space (zoom mode) this contradicts with "show as much information as possible without scrolling".
> 
> 
> 
> If I would make the lines closer together, they would be difficult to hit with the finger in normal mode.
> 
> 
> 
> The zoom mode would make the font so large that only 3 - 4 lines would be displayed.
> 
> 
> 
> It is difficult for the developers to satisfy everyone at the same time. Here you will have to live with a few compromises to handle different zoom sizes at the same time.
> 
> 
> 
> Translated with www.DeepL.com/Translator (free version)

Good points. Making best use of screen space when user / system can change text size is tricky and subjective. If not compromising function or usability, not urgent to address. 

If somehow one of our programming friends can spend some time thinking about screen layout optimization that works well both in Zoom and Standard display modes, would be a nice touch. Again, not urgent or high priority except where it causes a functional problem (like can't see / use controls).

**Comment by kagsteiner** [comment]
Concerning the change that the app now sometimes doesn’t show the dialog that allows me to enter a comment: sometimes this is quite unfortunate. Had the situation twice that I finished a match and couldn’t write something like “thanks for the match” in the end, which I believe is part of the DG netiquette. So I had to remember who I played with, go to the web UI, send a message there.

**Reply by hape42** [reply]
Yes, I agree with you. I've had that situation a few times now too, it doesn't feel good. 

So then it wasn't a solution for the annoying bug, but just a workaround. Damn. Then we have to go back to analyzing what exactly is happening in html and try to map that better in the app.

**Reply by hape42** [reply]
I have managed to recreate the situation. 

We have little to worry about. At this point, you can't chat at all. 

![Bildschirm­foto 2023-01-26 um 16 55 54](https://user-images.githubusercontent.com/2091581/214885978-af8d1733-0591-4992-b595-7b32e9d90b5a.png)

This is how it looks on the website. Sure, it can be done even better than the current solution. But there is no urgent need for action.

Or do you see it differently?

**Reply by kagsteiner** [reply]
Very unexpected. So DG has changed here? Very frustrating to build an app on something that is not an API but HTML code that can change any moment :-)

But that explains the sudden crashes... Are you in touch with the DG developer(s)? So you can get notice of changes or something?

**Reply by hape42** [reply]
>... Are you in touch with the DG developer(s)? So you can get notice of changes or something?

I had a brief conversation with Jordan about 4 years ago. It was only about whether he would mind if I made an IOS app. His "I'm sure people will like it. But he can't and won't help because of lack of time" was his last answer. All other questions about api or interface remained unanswered.  I understand that, he doesn't really want to invest any more time in the project except to keep it running. That's ok.  

Translated with www.DeepL.com/Translator (free version)

**Reply by hape42** [reply]
> ...So DG has changed here? Very frustrating to build an app on something that is not an API but HTML code that can change any moment :-)

I don't think DG has changed there. That has been the big problem for me. When you play a match, all kinds of HTML pages come up. There are a lot of possibilities with the "Next" button. Just play a few matches on the website and remember to watch what page comes up after each click. It is fascinatingly diverse.
Some of them come very rarely and are therefore also very difficult to implement in IOS.

End match with chat there is also still, which differs in HTML only very little.

There is still some work to be done here for us.


Translated with www.DeepL.com/Translator (free version)

**Comment by Smiles2go** [comment]
iOS 16.3 on iPhone 14pro has no apparent changes or new issues with DG build 366 (TestFlight).

**Reply by Smiles2go** [reply]
DG 1.4.4 Build 370 has been available in US App Store for several days now. Sorry - only just noticed.

**Comment by hape42** [comment]
done
